libs/vkd3d/resource.c

```diff
@@ -3493,6 +3493,9 @@ static void d3d12_descriptor_heap_write_null_descriptor_template(struct d3d12_de
     unsigned int offset;
     VkDeviceAddress *va;
 
+    if (!(desc->metadata.flags & VKD3D_DESCRIPTOR_FLAG_NON_NULL))
```

Comment:
I don't think this is correct for mutable_descriptor_type. When writing consecutive null descriptors, they can change type between two NULL descriptors. This seems correct for non-mutable however, so adding a check for that should be fine. Since this is pretty hot code, could be fine to add a bool to `heap->null_descriptor_template` to check if duplicate null descriptors should be elided.